sim/paladin/item_sets_pve.go

```diff
@@ -456,12 +456,40 @@ var ItemSetAvengersRadiance = core.NewItemSet(core.ItemSet{
 		},
 		4: func(agent core.Agent) {
 			paladin := agent.(PaladinAgent).GetPaladin()
-			paladin.OnSpellRegistered(func(spell *core.Spell) {
-				//"S03 - Item - TAQ - Paladin - Retribution 4P Bonus",
-				if spell.SpellCode == SpellCode_PaladinHolyWrath || spell.SpellCode == SpellCode_PaladinConsecration || spell.SpellCode == SpellCode_PaladinExorcism || spell.SpellCode == SpellCode_PaladinHolyShock || spell.SpellCode == SpellCode_PaladinHammerOfWrath {
-					// 4 Set: Increases the critical strike damage bonus of your Exorcism, Holy Wrath, Holy Shock, Hammer of Wrath, and Consecration by 60%.
-					spell.CritDamageBonus += 0.6
-				}
+
+			taq4pcAura := paladin.GetOrRegisterAura(core.Aura{
+				Label:     "Empower Exorcism",
+				ActionID:  core.ActionID{SpellID: 415073}, // temp id
+				Duration:  time.Second * 20,
+				MaxStacks: 3,
+				OnStacksChange: func(aura *core.Aura, sim *core.Simulation, oldStacks int32, newStacks int32) {
+					for _, exorcism := range paladin.exorcism {
+						exorcism.DamageMultiplierAdditive -= 0.4 * float64(oldStacks)
+						exorcism.DamageMultiplierAdditive += 0.4 * float64(newStacks)
+
+						if newStacks == 0 {
+							aura.Activate(sim)
+						}
+					}
+				},
+			})
+
+			paladin.RegisterAura(core.Aura{
+				Label:    "S03 - Item - TAQ - Paladin - Retribution 4P Bonus",
+				Duration: core.NeverExpires,
+				OnReset: func(aura *core.Aura, sim *core.Simulation) {
+					aura.Activate(sim)
+				},
+				OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if result.Landed() && spell.ProcMask.Matches(core.ProcMaskMeleeWhiteHit) {
+						if !taq4pcAura.IsActive() {
+							taq4pcAura.Activate(sim)
+						}
```

Comment:
Usually these kindss of effects will refresh every time, so you wouldn't want to make the `Activate` conditional